R/bias.R

```diff
@@ -247,71 +246,65 @@ bias_range <- function(range, lower, upper,
 #'
 #' Bias can assume values between
 #' -1 and 1 and is 0 ideally.
-#' @param predictions vector of length corresponding to the number of quantiles
-#' that holds predictions
-#' @param quantiles vector of corresponding size with the quantiles for which
-#' predictions were made
-#' @inheritParams bias_range
+#' @param lower vector of length corresponding to the number of central
+#' prediction intervals that holds predictions for the lower bounds of a
+#' prediction interval
+#' @param upper vector of length corresponding to the number of central
+#' prediction intervals that holds predictions for the upper bounds of a
+#' prediction interval
+#' @param range vector of corresponding size with information about the width
+#' of the central prediction interval
+#' @param true_value a single true value
 #' @return scalar with the quantile bias for a single quantile prediction
+#' @seealso bias_quantile bias_sample
 #' @author Nikos Bosse \email{nikosbosse@@gmail.com}
 #' @examples
 #'
-#' predictions <- c(
-#'   705.500, 1127.000, 4006.250, 4341.500, 4709.000, 4821.996,
-#'   5340.500, 5451.000, 5703.500, 6087.014, 6329.500, 6341.000,
-#'   6352.500, 6594.986, 6978.500, 7231.000, 7341.500, 7860.004,
-#'   7973.000, 8340.500, 8675.750, 11555.000, 11976.500
+#' lower <- c(
+#'   6341.000, 6329.500, 6087.014, 5703.500,
+#'   5451.000, 5340.500, 4821.996, 4709.000,
+#'   4341.500, 4006.250, 1127.000, 705.500
 #' )
 #'
-#' quantiles <- c(0.01, 0.025, seq(0.05, 0.95, 0.05), 0.975, 0.99)
+#' upper <- c(
+#'   6341.000, 6352.500, 6594.986, 6978.500,
+#'   7231.000, 7341.500, 7860.004, 7973.000,
+#'   8340.500, 8675.750, 11555.000, 11976.500
+#' )
+#'
+#' range <- c(0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 95, 98)
 #'
 #' true_value <- 8062
 #'
-#' bias_quantile(predictions, quantiles, true_value = true_value)
+#' bias_range(
+#'   lower = lower, upper = upper,
+#'   range = range, true_value = true_value
+#' )
 #' @export
 #' @keywords metric
 
-bias_quantile <- function(predictions, quantiles, true_value) {
-  # check that predictions and quantiles have the same length
-  if (!length(predictions) == length(quantiles)) {
-    stop("predictions and quantiles must have the same length")
-  }
+bias_range <- function(lower, upper, range, true_value) {
 
-  if (anyNA(predictions)) {
-    quantiles <- quantiles[!is.na(predictions)]
-    predictions <- predictions[!is.na(predictions)]
-  }
-  # if there is no input, return NA
-  if (length(quantiles) == 0 || length(predictions) == 0) {
-    return(NA_real_)
+  if (!length(range) > 1 && !all(diff(range) > 0)) {
```

Comment:
Should this be 
```
if (length(range) > 1 && !all(diff(range) > 0)) {
```
so without the exclamation mark? Maybe it would also be helpful to add a bracket

Reply:
Maybe also check that they're all in [0, 100]? Otherwise an error will be thrown in `bias_quantiles` which might be harder to interpret.

Reply:
> Maybe also check that they're all in [0, 100]? Otherwise an error will be thrown in bias_quantiles which might be harder to interpret.

I don't think we want to duplicate these checks do we or we end up where we were before with duplicate functions that are hard to maintain? We could make a little internal helper to avoid this with a scale and type arg (i.e. so it says either range or quantile).

Reply:
Both addressed (see 8f2c468)